src/models/sourcecodemodel.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+    SPDX-FileCopyrightText: Lieven Hey <lieven.hey@kdab.com>
+    SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#include "sourcecodemodel.h"
+#include "hotspot-config.h"
+
+#include <QFile>
+#include <QPalette>
+#include <QTextDocument>
+
+#ifdef KF5SyntaxHighlighting_FOUND
+#include <KSyntaxHighlighting/Definition>
+#include <KSyntaxHighlighting/Repository>
+#include <KSyntaxHighlighting/SyntaxHighlighter>
+#include <KSyntaxHighlighting/Theme>
+#endif
+
+SourceCodeModel::SourceCodeModel(QObject* parent)
+    : QAbstractTableModel(parent)
+    , m_document(new QTextDocument(this))
+#ifdef KF5SyntaxHighlighting_FOUND
+    , m_repository(std::make_unique<KSyntaxHighlighting::Repository>())
+    , m_highlighter(new KSyntaxHighlighting::SyntaxHighlighter(m_document))
+#endif
+{
+    qRegisterMetaType<QTextLine>();
```

Comment:
should be done in a central place, and ideally share code with the other model somehow

note that this doesn't necessarily mean a central base class, composition and extraction into a separate class would also work